File: tickpulse/backend/src/models/SubTask.js
const mysql = require('mysql2/promise');

const createSubtaskTable = async () => {
    try {
        // Connect to MySQL database
        const connection = await mysql.createConnection({
            host: process.env.MYSQL_HOST,
            user: process.env.MYSQL_USER,
            password: process.env.MYSQL_PASSWORD,
            database: process.env.MYSQL_DATABASE,
        });

        const createTableQuery = `
            CREATE TABLE IF NOT EXISTS SubTasks (
                id INT AUTO_INCREMENT PRIMARY KEY,
                task_id VARCHAR(50) NOT NULL,
                
                subtask_name VARCHAR(255),
                is_done TINYINT(1) DEFAULT 0,
                sort_order DOUBLE DEFAULT 0.0,

                FOREIGN KEY (task_id) REFERENCES Tasks(id) ON DELETE CASCADE
            ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
        `;

        await connection.execute(createTableQuery);
        console.log('Subtask table created or already exists.');
        await connection.end();
    } catch (error) {
        console.error('Error creating Subtask table:', error);
        throw error;
    }
};

module.exports = createSubtaskTable;

File: tickpulse/backend-go/internal/models/subtask.go
package models

type SubTask struct {
	ID          int     `db:"id" json:"id"`
	TaskID      string  `db:"task_id" json:"task_id"`
	SubtaskName *string `db:"subtask_name" json:"subtask_name"`
	IsDone      bool    `db:"is_done" json:"is_done"`
	SortOrder   float64 `db:"sort_order" json:"sort_order"`
}
